app/scripts/modules/core/src/api/mock/MockHttpClient.ts

```diff
@@ -0,0 +1,176 @@
+      this.predicates.url = (request) => !!url.exec(request.url);
+    } else if (typeof url === 'function') {
+      this.predicates.url = (request) => url(request.url);
+    }
+  }
+
+  public fulfilled = false;
+  private predicates: Record<string, IRequestPredicate> = {
+    verb: () => true,
+    url: () => true,
+    params: () => true,
+  };
+
+  isUnfulfilledMatch(request: IRequest) {
+    return !this.fulfilled && Object.values(this.predicates).every((predicate) => predicate(request));
+  }
+
+  response = {
+    status: 200,
+    data: null as any,
+  };
+
+  withParams(expectedParams: {}, exact = false): IExpectBuilder {
+    if (exact) {
+      this.predicates.params = (request) => isEqual(expectedParams, request.params);
+    } else {
+      this.predicates.params = (request) => isMatch(expectedParams, request.params);
+    }
+
+    return this;
+  }
+
+  respond(status: number, data?: any): IExpectBuilder {
+    this.response.status = status;
+    this.response.data = data;
+    return this;
+  }
+}
+
+interface IExpectBuilder {
+  withParams(expectedParams: {}, exact?: boolean): IExpectBuilder;
+  respond(status: number, data?: any): IExpectBuilder;
+}
+
+export class MockHttpClient implements IHttpClientImplementation {
+  public expectedRequests: ExpectRequest[] = [];
+  // public resolvedExpects: ExpectRequest[] = [];
+  public receivedRequests: IRequest[] = [];
+
+  public expect(verb: Verb, url?: UrlArg): IExpectBuilder {
+    const expected = new ExpectRequest(verb, url);
+    this.expectedRequests.push(expected);
+    return expected;
+  }
+
+  expectGET = (url?: string) => this.expect('GET', url);
+  expectPUT = (url?: string) => this.expect('PUT', url);
+  expectPOST = (url?: string) => this.expect('POST', url);
+  expectDELETE = (url?: string) => this.expect('DELETE', url);
+
+  request<T = any>(verb: Verb, url: string, params: {}): PromiseLike<T> {
+    const request: IRequest = { verb, url, params, expected: false };
+    this.receivedRequests.push(request);
+
+    const expectedRequest = this.expectedRequests.find((pendingExpect) => pendingExpect.isUnfulfilledMatch(request));
+    if (!expectedRequest) {
+      throw new Error(`Unexpected HTTP request: ${verb} ${url}`);
+    }
+
+    request.expected = true;
+    expectedRequest.fulfilled = true;
+
+    const { status, data } = expectedRequest.response;
+
+    if (status >= 200 && status <= 299) {
+      return Promise.resolve(data);
+    } else {
+      return Promise.reject({ status, data });
+    }
+  }
+
+  get = <T = any>(request: IRequest): PromiseLike<T> => this.request('GET', request.url, request.params);
+  put = <T = any>(request: IRequest): PromiseLike<T> => this.request('PUT', request.url, request.params);
+  post = <T = any>(request: IRequest): PromiseLike<T> => this.request('POST', request.url, request.params);
+  delete = <T = any>(request: IRequest): PromiseLike<T> => this.request('DELETE', request.url, request.params);
+
+  verifyNoOutstanding() {
```

Comment:
```suggestion
  verifyNoOutstandingExpectation() {
```
Might as well match the existing name to make migration easier /shrug